ipykernel/zmqshell.py

```diff
@@ -70,6 +78,15 @@ def _flush_streams(self):
         sys.stdout.flush()
         sys.stderr.flush()
 
+    @default('thread_local')
```

Comment:
Since I think this is the first PR that uses the traitlets 4.1 API, you may want to bump the version requirement in setup.py.